WordFrequency/WordFrequencyGame.cs

```diff
@@ -2,74 +2,69 @@
 using System.Collections.Generic;
 using System.Text.RegularExpressions;
 
-namespace WordFrequency
+namespace WordFrequency;
+
+public class WordFrequencyGame
 {
-    public class WordFrequencyGame
+    public string GetResult(string inputStr)
     {
-        public string GetResult(string inputStr)
-        {
-            if (Regex.Split(inputStr, @"\s+").Length == 1)
-            {
-                return inputStr + " 1";
-            }
-            else
-            {
-                //split the input string with 1 to n pieces of spaces
-                string[] arr = Regex.Split(inputStr, @"\s+");
 
-                List<Input> inputList = new List<Input>();
-                foreach (var s in arr)
-                {
-                    Input input = new Input(s, 1);
-                    inputList.Add(input);
-                }
+        string[] splitedInput = SplitInputString(inputStr);
 
-                //get the map for the next step of sizing the same word
-                Dictionary<string, List<Input>> map = GetListMap(inputList);
+        List<WordCount> wordCounts = splitedInput.Select(s => new WordCount(s, 1)).ToList(); ;
 
-                List<Input> list = new List<Input>();
-                foreach (var entry in map)
-                {
-                    Input input = new Input(entry.Key, entry.Value.Count);
-                    list.Add(input);
-                }
 
-                inputList = list;
 
-                inputList.Sort((w1, w2) => w2.WordCount - w1.WordCount);
+        List<WordCount> resultList = TransformToOutputList(wordCounts);
 
-                List<string> strList = new List<string>();
+        List<string> strList = resultList.Select(wc => $"{wc.Word} {wc.Count}").ToList();
 
-                //stringJoiner joiner = new stringJoiner("\n");
-                foreach (Input w in inputList)
-                {
-                    string s = w.Value + " " + w.WordCount;
-                    strList.Add(s);
-                }
+        return string.Join("\n", strList);
 
-                return string.Join("\n", strList.ToArray());
-            }
+    }
+
+    private static string[] SplitInputString(string inputStr)
+    {
+        string[] arr = Regex.Split(inputStr, @"\s+");
+        if (arr.Length == 1)
+        {
+            return new string[] { $"{arr[0]} 1" };
         }
 
-        private Dictionary<string, List<Input>> GetListMap(List<Input> inputList)
+        return arr;
+
+    }
+
+    private List<WordCount> TransformToOutputList(List<WordCount> wordCounts)
+    {
+        Dictionary<string, int> wordCountMap = GetWordCountMap(wordCounts);
+
+
+        List<WordCount> resultList = wordCountMap.Select(entry => new WordCount(entry.Key, entry.Value)).ToList();
+
+        resultList.Sort((w1, w2) => w2.Count - w1.Count);
+        return resultList;
+    }
+
+
+    private Dictionary<string, int> GetWordCountMap(List<WordCount> wordCounts)
```

Comment:
return type is a dictionary and also contain sort function, maybe could change method name to "GetSortedWordCountDict"